pkgs/servers/home-assistant/patches/static-symlinks.patch

```diff
@@ -0,0 +1,37 @@
+diff --git a/homeassistant/components/frontend/__init__.py b/homeassistant/components/frontend/__init__.py
+index c6812f4d9d..64257c4edb 100644
+--- a/homeassistant/components/frontend/__init__.py
++++ b/homeassistant/components/frontend/__init__.py
+@@ -358,7 +358,7 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+ 
+     local = hass.config.path("www")
+     if os.path.isdir(local):
+-        hass.http.register_static_path("/local", local, not is_dev)
++        hass.http.register_static_path("/local", local, not is_dev, follow_symlinks=True)
+ 
+     hass.http.app.router.register_resource(IndexView(repo_path, hass))
+ 
+diff --git a/homeassistant/components/http/__init__.py b/homeassistant/components/http/__init__.py
+index 9e77563f7a..438a879038 100644
+--- a/homeassistant/components/http/__init__.py
++++ b/homeassistant/components/http/__init__.py
+@@ -305,16 +305,16 @@ class HomeAssistantHTTP:
+         )
+ 
+     def register_static_path(
+-        self, url_path: str, path: str, cache_headers: bool = True
++        self, url_path: str, path: str, cache_headers: bool = True, follow_symlinks: bool = False
+     ) -> None:
+         """Register a folder or file to serve as a static path."""
+         if os.path.isdir(path):
+             if cache_headers:
+                 resource: CachingStaticResource | web.StaticResource = (
+-                    CachingStaticResource(url_path, path)
++                    CachingStaticResource(url_path, path, follow_symlinks=follow_symlinks)
```

Comment:
Are you thinking of upstreaming this?  I wonder if there are any security implications.

Reply:
https://github.com/home-assistant/core/pull/65191 was closed as too heated. :shrug:

Reply:
Didn't seem that heated!  It would be nice if this was at least an option, but I guess we can't even comment on it now.

Given it's controversial, perhaps it's better that we only enable the patch if there are some lovelace modules configured, and document that it turns this behaviour on?

Reply:
No, it wasn't heated at all. This is about burned ground.

No strong opinion about making the patch conditional. I'd think our hardening would contain most problems.